strawberry/starlite/controller.py

```diff
@@ -0,0 +1,399 @@
+                for supported_header in ("text/html", "*/*")
+            )
+
+        def get_graphiql_response(self) -> Response[str]:
+            html = get_graphiql_html()
+            return Response(html, media_type=MediaType.HTML)
+
+        @staticmethod
+        def _merge_responses(
+            response: Response, actual_response: Response
+        ) -> Response[Union[GraphQLResource, str]]:
+            actual_response.headers.update(response.headers)
+            actual_response.cookies.extend(response.cookies)
+            actual_response.background = response.background
+            if response.status_code:
+                actual_response.status_code = response.status_code
+
+            return actual_response
+
+        @get(
+            responses={
+                HTTP_404_NOT_FOUND: ResponseSpec(
+                    description="Not found if GraphiQL is not enabled.",
+                    model=EmptyResponseModel,
+                    media_type=MediaType.TEXT,
+                ),
+                HTTP_400_BAD_REQUEST: ResponseSpec(
+                    description="Unable to parse request body as JSON",
+                    model=EmptyResponseModel,
+                    media_type=MediaType.TEXT,
+                ),
+            },
+        )
+        async def handle_http_get(
+            self,
+            request: Request,
+            context: CustomContext,
+            root_value: Any,
+        ) -> Response[Union[GraphQLResource, str]]:
+            actual_response: Response
+
+            if request.query_params:
+                try:
+                    query_data = parse_query_params(
+                        cast("Dict[str, Any]", request.query_params)
+                    )
+                except json.JSONDecodeError:
+                    return Response(
+                        "Unable to parse request body as JSON",
+                        status_code=HTTP_400_BAD_REQUEST,
+                        media_type=MediaType.TEXT,
+                    )
+                return await self.execute_request(
+                    request=request,
+                    data=query_data,
+                    context=context,
+                    root_value=root_value,
+                )
+            if self.should_render_graphiql(request):
+                return cast(
+                    "Response[Union[GraphQLResource, str]]",
+                    self.get_graphiql_response(),
+                )
+            return Response(content="Bad request", status_code=HTTP_404_NOT_FOUND)
+
+        @post(status_code=HTTP_200_OK)
+        async def handle_http_post(
+            self,
+            request: Request,
+            context: CustomContext,
+            root_value: Any,
+        ) -> Response[Union[GraphQLResource, str]]:
+            actual_response: Response[Union[GraphQLResource, str]]
+
+            content_type = request.headers.get("content-type", "")
```

Comment:
You can access `request.content_dusposition`, which is a tuple where the first element is the content type header